### ISSUE TITLE
[Bug]: Adapt jest-cli path based on if CJS or ESM

### ISSUE THREAD

**Issue by Aghassi** [aspect-build/rules_jest#69]
### What happened?

Today, the _test.sh file generated assumes you are running in a common js world. It hard codes `jest-cli/bin/index.js` as the entry point no matter what. If you run this file when doing esm based runs, you will fail with error 
```
SyntaxError: Unexpected token ','
    at Loader.moduleStrategy (node:internal/modules/esm/translators:146:18)
    at async link (node:internal/modules/esm/module_job:67:21)
================================================================================
```

What should happen is the following should references the `main` path in `package.json` when esm is detected, not the `bin` path

```
    entry_point="../jest/node_modules/.aspect_rules_js/jest-cli@28.1.0/node_modules/jest-cli/main/jest.js"
    if [[ "$entry_point" == ../* ]]; then
        entry_point="external/${entry_point:3}"
    fi
    entry_point="$JS_BINARY__EXECROOT/$BAZEL_BINDIR/$entry_point"
else
    entry_point="$JS_BINARY__RUNFILES/rh/../jest/node_modules/.aspect_rules_js/jest-cli@28.1.0/node_modules/jest-cli/bin/jest.js"
```

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`:
bazel 6.0.0-pre.20221020.1

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
HEAD

Language(s) and/or frameworks involved:Typescript/Jest

### How to reproduce

```shell
define jest_test



    jest_test(
        name = "%s_test" % name,
        config = "//web/web-app/app:jest_config",
        data = [
            ":%s_transpile_test_files" % name,
            "//web/web-app/app:package_json"
        ],
        node_options = [
            "--experimental-vm-modules",
        ]
    )
```

Add `type: module` to `package.json
Use `jest.config.mjs` as input
Pass in ESM output from `ts_project`

Then run the test on Node 16. It _should_ fail
```


### Any other information?

_No response_

### Fund our work

- [ ] Sponsor our open source work by donating a [bug bounty](https://opencollective.com/aspect-build/)

**Comment by flolu** [comment]
I was able to fix this issue by updating to the latest Node.js version.

Before:

```
load("@rules_nodejs//nodejs:repositories.bzl", "nodejs_register_toolchains")

nodejs_register_toolchains(
    name = "nodejs",
    node_version = "16.9.0",
)
```

Afterr:

```
load("@rules_nodejs//nodejs:repositories.bzl", "DEFAULT_NODE_VERSION", "nodejs_register_toolchains")

nodejs_register_toolchains(
    name = "nodejs",
    node_version = DEFAULT_NODE_VERSION,
)
```

**Comment by Aghassi** [comment]
@flolu Which version of Node does that default to?

**Comment by flolu** [comment]
@Aghassi I think `"16.19.0"` (https://github.com/bazelbuild/rules_nodejs/blob/stable/docs/Core.md#node_version)

**Comment by jbedard** [comment]
@Aghassi does using a more recent version of node solve this?

**Comment by Aghassi** [comment]
I believe we went from 16.15 to 16.18 and it solved this. It's no longer an issue for us and I think @gregmagolan also added some fixes.
